From 91f7b1eb1204c5edd17707042f281c419bf123aa Mon Sep 17 00:00:00 2001
From: Keith Kyzivat <keith.kyzivat@qt.io>
Date: Tue, 11 Jun 2024 00:11:06 -0400
Subject: [PATCH] Fix separately built QScintilla

When building QScintilla separately, rather than just picking up the
distribution-installation of QScintilla, some things were out of date
and needed updating. This commit updates those.

* Source URL changed
* QScintilla outside of the Qt dir wasn't recognized, added recognition
  outside of Qt.
* a couple of other small things
---
 CMakeLists.txt                        | 6 ++++++
 cmake/Modules/FindQt5QScintilla.cmake | 4 ++--
 scripts/uni-build-dependencies.sh     | 8 +++++---
 3 files changed, 13 insertions(+), 5 deletions(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index 7b6c842af..ecd15aa10 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -556,6 +556,9 @@ if(NOT HEADLESS)
     endif()
 
     find_package(Qt6QScintilla 2.8.0 REQUIRED QUIET)
+    message(STATUS "QT6QSCINTILLA_LIBRARY: ${QT6QSCINTILLA_LIBRARY}")
+    message(STATUS "QT6QSCINTILLA_INCLUDE_DIRS: ${QT6QSCINTILLA_INCLUDE_DIRS}")
+    target_include_directories(OpenSCAD PRIVATE "${QT6QSCINTILLA_INCLUDE_DIRS}/..")
     message(STATUS "QScintilla: ${QT6QSCINTILLA_VERSION_STRING}")
 
     if(ENABLE_QTDBUS)
@@ -585,6 +588,9 @@ if(NOT HEADLESS)
     endif()
 
     find_package(Qt5QScintilla 2.8.0 REQUIRED QUIET)
+    message(STATUS "QT5QSCINTILLA_LIBRARY: ${QT5QSCINTILLA_LIBRARY}")
+    message(STATUS "QT5QSCINTILLA_INCLUDE_DIRS: ${QT5QSCINTILLA_INCLUDE_DIRS}")
+    target_include_directories(OpenSCAD PRIVATE "${QT5QSCINTILLA_INCLUDE_DIRS}/..")
     message(STATUS "QScintilla: ${QT5QSCINTILLA_VERSION_STRING}")
 
     if(ENABLE_QTDBUS)
diff --git a/cmake/Modules/FindQt5QScintilla.cmake b/cmake/Modules/FindQt5QScintilla.cmake
index d25195d2c..d9a1dd114 100644
--- a/cmake/Modules/FindQt5QScintilla.cmake
+++ b/cmake/Modules/FindQt5QScintilla.cmake
@@ -45,7 +45,7 @@
 
 find_path ( QT5QSCINTILLA_INCLUDE_DIR
   NAMES qsciscintilla.h
-  HINTS ${Qt5Widgets_INCLUDE_DIRS}
+  HINTS ${Qt5Widgets_INCLUDE_DIRS} ${OPENSCAD_DEPS}/include
   PATH_SUFFIXES qt5 Qsci
 )
 
@@ -81,7 +81,7 @@ endif ()
 
 find_library ( QT5QSCINTILLA_LIBRARY
   NAMES qt5scintilla2 qscintilla2-qt5 qscintilla2 qscintilla2_qt5
-  HINTS ${Qt5Widgets_LIBRARIES}
+  HINTS ${Qt5Widgets_LIBRARIES} ${OPENSCAD_DEPS}/lib
 )
 
 set ( QT5QSCINTILLA_LIBRARIES ${QT5QSCINTILLA_LIBRARY} )
diff --git a/scripts/uni-build-dependencies.sh b/scripts/uni-build-dependencies.sh
index b16a26ac1..0de1a9563 100755
--- a/scripts/uni-build-dependencies.sh
+++ b/scripts/uni-build-dependencies.sh
@@ -157,10 +157,10 @@ build_qt5scintilla2()
   cd $BASEDIR/src
   rm -rf ./QScintilla-gpl-$version.tar.gz
   if [ ! -f QScintilla-gpl-$version.tar.gz ]; then
-     curl -L -o "QScintilla-gpl-$version.tar.gz" "http://downloads.sourceforge.net/project/pyqt/QScintilla2/QScintilla-$version/QScintilla-gpl-$version.tar.gz?use_mirror=switch"
+     curl -L -o "QScintilla-gpl-$version.tar.gz" "https://sourceforge.net/projects/pyqt/files/QScintilla2/QScintilla-$version/QScintilla_gpl-$version.tar.gz/download#"
   fi
   tar xzf QScintilla-gpl-$version.tar.gz
-  cd QScintilla-gpl-$version/Qt4Qt5/
+  cd QScintilla_gpl-$version/Qt4Qt5/
   qmake CONFIG+=staticlib
   tmpinstalldir=$DEPLOYDIR/tmp/qsci$version
   INSTALL_ROOT=$tmpinstalldir make -j"$NUMCPU" install
@@ -174,9 +174,11 @@ build_qt5scintilla2()
   if [ ! -e $DEPLOYDIR/include/Qsci ]; then
     # workaround numerous bugs in qscintilla build system, see 
     # ../qscintilla2.prf and ../scintilla.pri for more info
-    qsci_staticlib=`find $tmpinstalldir -name libqscintilla2.a`
+    qsci_staticlib=`find $tmpinstalldir -name libqscintilla2_qt5.a`
     qsci_include=`find $tmpinstalldir -name Qsci`
     if [ -e $qsci_staticlib ]; then
+      mkdir -p $DEPLOYDIR/include
+      mkdir -p $DEPLOYDIR/lib
       cp -av $qsci_include $DEPLOYDIR/include/
       cp -av $qsci_staticlib $DEPLOYDIR/lib/
     else
-- 
2.40.1

